app/scripts/metamask-controller.js

```diff
@@ -2075,7 +2071,13 @@ export default class MetamaskController extends EventEmitter {
     checkForMultipleVersionsRunning();
   }
 
-  triggerNetworkrequests() {
+  async triggerNetworkrequests() {
+    // TODO: Delete when ready to remove `networkVersion` from provider object
+    this.deprecatedNetworkId = null;
+    this.updateDeprecatedNetworkId();
+
+    await this.networkController.lookupNetwork();
```

Comment:
Previously, this only got called once, when initializing the extension. Will this get called every time MetaMask is opened now?

Reply:
Rather than "initializing" minus the lookup during startup, then repeating the lookup each time the UI is open, perhaps it would be simpler to delay the initialization altogether.

We could have a one-time initialization function that gets called post-onboarding, just one time. Then we get rid of these repeat lookups, and we no longer need a patch.

Reply:
@Gudahtt I am finally coming back to this. The problem is that we need a block tracker and provider to initialize a few other controllers. So if we postpone initialization of the network controller entirely (instead of just postponing the `lookup`), then also have to postpone initialization of a those other controllers. Is that what you are suggesting?

Reply:
Nope, I was suggesting that we delay just the lookup operation. We don't need to perform a lookup to create a provider and block tracker.